api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/BaseReferenceHandlingTranslator.java

```diff
@@ -41,45 +41,35 @@
 	public static final String TEST_ORDER_TYPE_UUID = "52a447d3-a64a-11e3-9aeb-50e549534c5e";
 	
 	protected Reference createEncounterReference(@NotNull Encounter encounter) {
-		if (encounter instanceof HibernateProxy) {
```

Comment:
I think I have a nicer solution than this merged into the code base. I.e., we can remove all of these checks.